### ISSUE TITLE
Show error when combining `-t` and `-p` cli options

### ISSUE THREAD

**Issue by djbe** [SwiftGen/SwiftGen#283]
Right now, a user can invoke swiftgen using something like:
```swiftgen -p their-custom-template.stencil -t swift3 ...```

This might confuse the user as to the purpose of either option, or a behaviour they might not expect. We should throw an error if both options are present.

**Comment by Liquidsoul** [comment]
I am not sure if this is really an _easy_ task. But maybe I am missing some information.

As I understand it, we are computing the resolved template in the [`findTemplate`](https://github.com/SwiftGen/SwiftGen/blob/d79f8a34c62cfe3c6a672a7e99948980d27172f3/Sources/ArgumentsUtils.swift#L128-L145).
Its arguments comes from `Commander`'s `Option` which means we do not have the information at this point if the value was set or if we are getting the default value.
Moreover, it is impossible for us to know if the user has set the `-t` option to its default value or not.

TL;DR:
- we cannot know it the user invoked swiftgen with `-t default` or without a `-t` option
- information about the `Option`s default values must be passed down to `findTemplate` in _every_ `command` call we have.

**Comment by AliSoftware** [comment]
@Liquidsoul we plan to remove the"fallback to default template" behavior in 5.0. Both because it means changing the default when a new Swift version is released — but it's not reasonable to change the default without notifying the user and last old commands relying on one default template suddenly change the template in the next version because we decided to change the default — and because avoiding the fallback will force people to specify explicitly which template they want and make them realize they have three choice and that this template mechanism exists (we had too many issues in the past from people claiming SwiftGen didn't support Swift 3 because they didn't realize you could choose alternate templates) 

So that should solve the issue you're mentioning I guess.

**Comment by AliSoftware** [comment]
Closed by #300
